### PR TITLE
Use PHP 7.4 for WooCommerce Blocks' `wp-env` Config

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Setting the environment's default PHP version to WooCommerce's minimum will prevent accidentally introducing incompatible code.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm --filter='@woocommerce/block-library' env:restart`
2. Navigate to the SSR and confirm that the PHP version is 7.4.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Just a change to the dev environment config.

</details>

### CODE REVIEW

**Review by samueljseay — APPROVED** [review]
Changes look good! Tested well! 🚢

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php [hunk 1]
```diff
@@ -304,14 +304,16 @@ private function enable_client_side_navigation( $block_content ) {
 			$p->set_attribute(
 				'data-wc-context',
 				wp_json_encode(
-					array(
-						...$current_context,
```
Comment (roykho) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php:308]: Wasn't this introduced in 7.4?

Comment (ObliviousHarmony) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php:308]: While the splat operator _was_ introduced in 7.4, [you couldn't use it on associative arrays until PHP 8.1](https://www.php.net/releases/8.1/en.php#array_unpacking_support_for_string_keyed_arrays)!

Comment (imanish003) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php:308]: Thanks for catching this @ObliviousHarmony 💜

**Pull request merged**